operator/cmd/mesh/manifest-migrate.go

```diff
@@ -105,12 +109,21 @@ func translateFunc(values []byte, l *Logger) error {
 	}
 
 	// Not IstioControlPlane, try Helm values.yaml.
-	ts, err := translate.NewReverseTranslator(binversion.OperatorBinaryVersion.MinorVersion)
+	mvs := binversion.OperatorBinaryVersion.MinorVersion
+	ts, err := translate.NewReverseTranslator(mvs)
 	if err != nil {
 		return fmt.Errorf("error creating values.yaml translator: %s", err)
 	}
 
-	translatedIOPS, err := ts.TranslateFromValueToSpec(values)
+	// verify the input schema first
+	if errs := validate.CheckValuesString(values); len(errs) != 0 {
+		vs := fmt.Sprintf("releaese-%s.%d", mvs.MajorVersion, mvs.Minor)
+		return fmt.Errorf("the input values.yaml fail validation: %v \n"+
```

Comment:
this error text should be generated in the helper function. top level functions should simply pass through lower level errors, the only exception would be if decorations are needed for extra context you can't get in the lower level function. in this case, the error message is completely in the domain of validate pkg and not here - this should just be return err.

Reply:
I agree that the first line would be in the low level validate function, but the second and third line should be here. One reason is that it is a usage guidance about this command which is not applied to all cases when calling the validation function. Another reason is that the low level package does not depend on specific version(it probably should but at least now our values.proto is not versioned)

Reply:
all right, fair enough :-) in that case could you please remove the first line and and put the big string in a const somewhere? we seem to be repeating the same help message all over the place with slight variations in the text. my intent here is also to try to make the function more compact and readable. thanks!